src/lex.cpp

```diff
@@ -1241,6 +1244,23 @@ void lexer::skip_block_comment() {
   this->input_ += strlen(this->input_);
 }
 
+void lexer::skip_html_line_comment() {
+  QLJS_ASSERT(this->input_[0] == '<' && this->input_[1] == '!' &&
+              this->input_[2] == '-' && this->input_[3] == '-');
+  for (char8* c = this->input_ + 4;; ++c) {
+    int newline_size = this->newline_character_size(c);
+    if (newline_size > 0) {
+      this->input_ = c + newline_size;
+      this->skip_whitespace();
+      break;
+    }
+    if (*c == u8'\0' && this->is_eof(c)) {
+      this->input_ = c;
+      break;
+    }
```

Comment:
Minor: Hmm, is this loop body exactly the same as `lexer::skip_line_comment`'s? Perhaps we should factor the common code. Maybe later.

Reply:
Added the check for `<!--` to `skip_line_comments()`
and adjusted `skip_length`.